File: kutil-unit-api/src/main/java/net/kigawa/kutil/unitapi/annotation/Dependency.java
package net.kigawa.kutil.unitapi.annotation;

import java.lang.annotation.*;

@Repeatable(value = Dependencies.class)
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface Dependency
{
    Class<?> value();

    String name() default "";
}

File: kunit/src/jvmMain/kotlin/net/kigawa/kutil/kunit/api/annotation/Dependency.kt
package net.kigawa.kutil.kunit.api.annotation

import kotlin.reflect.KClass

@JvmRepeatable(value = Dependencies::class)
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class Dependency
  (val value: KClass<*>, val name: String = "")
